test/Streamly/Test/Data/Parser/ParserD.hs

```diff
@@ -178,6 +178,32 @@ takeBetween =
                             else property False
                         Left _ -> property (m > n || list_length < m)
 
+-- deintercalate :: Property
+-- deintercalate =
+--     forAll (chooseInt (min_value, max_value)) $ \n ->
+--         forAll (listOf (chooseInt (min_value, max_value))) $ \ls ->
+--             case S.parseD (P.deintercalate FL.drain (P.satisfy odd) FL.drain
+--                 (P.satisfy even)) (S.fromList ls)  of
+--                 Right parsed_list -> property (length ls == length parsed_list)
+--                 Left _ -> property False
+
+deintercalate :: Property
+deintercalate =
```

Comment:
This is being tested in `Parser` as well. I guess you can skip testing this here.